### PR TITLE
copy more utilities from dartlang to atom.dart

### PR DESCRIPTION
**Pull request by danrubel**
@devoncarew

### CODE REVIEW

File: lib/atom_utils.dart [hunk 1]
```diff
@@ -69,3 +70,29 @@ class TrustedHtmlTreeSanitizer implements NodeTreeSanitizer {
...
 }
+
+/// Look for the given executable; throw an error if we can't find it.
+///
+/// Note: on Windows, this assumes that we're looking for an `.exe` unless
+/// `isBatchScript` is specified.
+Future<String> which(String execName, {bool isBatchScript: false}) {
+  if (isMac) {
+    // /bin/bash -l -c 'which dart'
+    String shell = process.env('SHELL') ?? '/bin/bash';
```
Comment (devoncarew) [inline lib/atom_utils.dart:81]: I'll need to do some surgery in here in a day or two -

**Comment by devoncarew** [comment]
lgtm

File: lib/atom_utils.dart (hunk 1 above)
Comment (danrubel) [inline lib/atom_utils.dart:81]: I'll get this committed ASAP then refactor dartlang to use it.

Comment (danrubel) [inline lib/atom_utils.dart:81]: https://github.com/dart-atom/dartlang/pull/804

**Pull request merged**